app/lib/forms/work_setting.rb

```diff
@@ -0,0 +1,51 @@
+module Forms
+  class WorkSetting < Base
+    VALID_WORK_SETTING_OPTIONS = %w[early_years_or_childcare a_school an_academy_trust a_16_to_19_educational_setting other].freeze
+
+    attr_accessor :work_setting
+
+    validates :work_setting, presence: true, inclusion: { in: VALID_WORK_SETTING_OPTIONS }
+
+    def self.permitted_params
+      %i[work_setting]
+    end
+
+    def after_save
+      # we are inferring `works_in_school` and `works_in_childcare` to maintain
+      # consistency with older records
+
+      case work_setting
+      when "a_school", "an_academy_trust", "a_16_to_19_educational_setting"
+        wizard.store["works_in_school"] = "yes"
+        wizard.store["works_in_childcare"] = "no"
+        wizard.store["works_in_nursery"] = nil
```

Comment:
This is such a nice side-effect of these changes. That having these two major branches determined in one place means we can reliably clean up data related to related answers.

There's also kind_of_nursery and has_ofsted_urn that we could clean up in situations where the user says they work in a school. And off of that we could possibly clear institution_name and institution_identifier if we check that the work_setting has changed from one to the other.

Reply:
Ah yeah - I wondered about that but it looked like them still being set didn't have any side effects, even if they're unused.

Reply:
I've pushed I've pushed [1b36056](https://github.com/DFE-Digital/npq-registration/pull/475/commits/1b3605618ced556b3a01180956c57bdea1a9c317) that also removes `kind_of_nursery` and `has_ofsted_urn` - but it removes the keys rather than sets them as `nil`. This should make testing a bit more straightforward. that also removes `kind_of_nursery` and `has_ofsted_urn` - but it removes the keys rather than sets them as `nil`. This should make testing a bit more straightforward.